keps/sig-api-machinery/4355-coordinated-leader-election/README.md

```diff
@@ -440,6 +451,91 @@ set of candidates and selected strategy is the same as before.
 The obvious drawback is the need for a consensus protocol and extra information
 in the `LeaseCandidate` object that may be unnecessary.
 
+### Priority-based Coordinated Leader Election
+To enhance control over leader assignment beyond existing CLE strategies like OldestEmulatedVersion, we propose adding an optional `Priority` field unset by default, higher value = higher priority) to `LeaseCandidateSpec`.
+
+This field allows operators to explicitly designate a preferred leader.
+The CLE system will select the candidate with the highest non-zero Priority. If multiple candidates share the same highest priority, the existing v1.CoordinatedLeaseStrategy will act as a tie-breaker. If no candidates have a priority set, the system defaults to the existing v1.CoordinatedLeaseStrategy. 
+
+This provides granular, temporary control without replacing the primary CLE mechanism.
+
+#### LeaseCandidateSpec Update
+A new field called `Priority` is included into `LeaseCandidateSpec`:
+```go
+// LeaseCandidateSpec is a specification of a Lease.
+type LeaseCandidateSpec struct {
+	// ...
+  Priority *int32 `json:"priority,omitempty" protobuf:"varint,7,opt,name=priority"` // New field: Higher value means higher priority. Default 0 (no explicit priority).
+}
+```
+
+#### Behavior of the Priority Field
+- Priority Value: The `Priority` field is an point to int32. A higher numerical value indicates a higher priority. This field must be greater than 0.
+- Default Value: If a `LeaseCandidateSpec` does not have the Priority field set, or if it is set to 0, it is considered to have no explicit priority preference.
+- Selection Logic:
+  - If one or more candidates have a Priority > 0: The candidate with the numerically highest Priority value will be selected as the leader.
```

Comment:
If we only care about values greater than zero, the field can be a non-pointer with omitempty